pkg/sql/gcjob/index_garbage_collection.go

```diff
@@ -40,10 +40,28 @@ func deleteIndexData(
 		log.Infof(ctx, "GC is being considered on table %d for indexes indexes: %+v", parentID, droppedIndexes)
 	}
 
+	maybeHandleDeletedDescriptor := func(err error) (done bool) {
+		// If the descriptor has been removed, then we need to assume that the relevant
+		// zone configs and data have been cleaned up by another process.
+		if !errors.Is(err, catalog.ErrDescriptorNotFound) {
+			return false
+		}
+		log.Infof(ctx, "descriptor %d dropped, assuming another process has handled GC", parentID)
+		for _, index := range droppedIndexes {
+			markIndexGCed(
+				ctx, index.IndexID, progress, jobspb.SchemaChangeGCProgress_CLEARED,
+			)
+		}
+		return true
+	}
+
 	// Before deleting any indexes, ensure that old versions of the table descriptor
 	// are no longer in use. This is necessary in the case of truncate, where we
 	// schedule a GC Job in the transaction that commits the truncation.
 	parentDesc, err := sql.WaitToUpdateLeases(ctx, execCfg.LeaseManager, parentID)
+	if maybeHandleDeletedDescriptor(err) {
```

Comment:
I feel like a eaiser to understand implementation is
```go
		if errors.Is(err, catalog.ErrDescriptorNotFound) {
		  // If the descriptor has been removed, then we need to assume that the relevant
		  // zone configs and data have been cleaned up by another process.
			handleDeletedDescriptor()
      return
		}
```
and accordingly, change the other function to
```go
	handleDeletedDescriptor := func() {
		log.Infof(ctx, "descriptor %d dropped, assuming another process has handled GC", parentID)
		for _, index := range droppedIndexes {
			markIndexGCed(
				ctx, index.IndexID, progress, jobspb.SchemaChangeGCProgress_CLEARED,
			)
		}
		return true
	}
```

But I don't really have a strong opinion on this; either way is perfectly fine with me.